cmd/ferretdb/main.go

```diff
@@ -111,20 +119,47 @@ func run() {
 		return
 	}
 
-	startFields := []zap.Field{
+	stateFile, err := filepath.Abs(filepath.Join(cli.StateDir, "state.json"))
+	if err != nil {
+		log.Fatalf("Failed to get path for state file: %s.", err)
+	}
+
+	p, err := state.NewProvider(stateFile)
+	if err != nil {
+		log.Fatalf("Failed to create state provider: %s.", err)
+	}
+
+	s, err := p.Get()
+	if err != nil {
+		log.Fatalf("Failed to get state: %s.", err)
+	}
+
+	level, err := zapcore.ParseLevel(cli.Log.Level)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	logUUID := s.UUID
+	startupFields := []zap.Field{
 		zap.String("version", info.Version),
 		zap.String("commit", info.Commit),
 		zap.String("branch", info.Branch),
 		zap.Bool("dirty", info.Dirty),
 		zap.Bool("debug", info.Debug),
 		zap.Reflect("buildEnvironment", info.BuildEnvironment.Map()),
 	}
-	logger.Info("Starting FerretDB "+info.Version+"...", startFields...)
 
-	if !slices.Contains(clientconn.AllModes, clientconn.Mode(cli.Mode)) {
-		logger.Sugar().Fatalf("Unknown mode %q.", cli.Mode)
+	// don't add UUID to all messages, but log it once at startup
+	if !cli.Log.UUID {
```

Comment:
So we did all that statefile-related stuff to just learn that `cli.Log.UUID == false` and we don't need anything from it... I mean, I guess we are going to use statefile for something else as well, but still it looks a bit strange. Don't you want to encapsulate all the statefile/telemetry/etc related things to at least a separate function to simplify readability?